Blackjack/Game/BlackjackGame.m

```diff
@@ -0,0 +1,262 @@
+        return;
+      }
+      break;
+    }
+      
+    case SURRENDER:
+      [self.currentPlayer setState:BUST];
+      
+    case STAND: {
+      break;
+    }
+  }
+  
+  if (isLastPlayer) {
+    [self finlizeRound];
+  } else {
+    [self nextPlayer];
+  }
+}
+
+- (void)collectBet:(NSInteger)amount {
+  if (self.currentPlayer.chips >= amount) {
+    [self.bets setObject:[NSNumber numberWithInteger:amount] forKey:self.currentPlayer.identifier];
+    [self.currentPlayer collectBet:amount];
+    [self nextPlayer];
+  } else {
+    NSAssert(self.currentPlayer.chips < amount, @"Player should never get to this point");
+  }
+  
+  if (self.bets[self.players.lastObject.identifier]) {
+    [self dealCardsFaceUp:YES];
+    [self dealCardsFaceUp:NO];
+    
+    if (self.currentPlayer == self.players.firstObject && self.currentPlayer.state != PLAYING) {
+      [self nextPlayer];
+    }
+    self.state = BETS_OVER;
+  }
+}
+
+-(void)dealCardsFaceUp:(BOOL)isFaceUp {
+  for (Player *player in self.players) {
+    BOOL _isFaceUp = player == self.players.firstObject ? YES : isFaceUp;
+    [player acceptNewCard:[self.deck drawRandomCardWithFaceUp:_isFaceUp]];
+  }
+  [self.dealer acceptNewCard:[self.deck drawRandomCardWithFaceUp:isFaceUp]];
+}
+
+-(void)nextPlayer {
+  NSMutableArray<NSNumber *> *playersIndexToUpdated = [NSMutableArray new];
+  
+  if (self.currentPlayer != self.players.lastObject) {
+    NSNumber *currentPlayerIndex = [NSNumber numberWithInteger:[self.players indexOfObject: self.currentPlayer]];
+    [playersIndexToUpdated addObject:currentPlayerIndex];
+    
+    Player *nextPlayer = self.players[currentPlayerIndex.intValue + 1];
+    [nextPlayer showHand];
+    self.currentPlayer = nextPlayer;
+    
+    NSNumber *nextPlayerIndex = [NSNumber numberWithInteger:[self.players indexOfObject: nextPlayer]];
+    [playersIndexToUpdated addObject:nextPlayerIndex];
+    
+    [self.delegate focusOnPlayerAtIndex:playersIndexToUpdated.lastObject.integerValue];
+  } else {
+    NSArray<Player *> *currentActivePlayers = [self currentActivePlayers];
+    
+    self.currentPlayer = currentActivePlayers.firstObject;
+    [playersIndexToUpdated addObject:[NSNumber numberWithInteger:0]];
+    [playersIndexToUpdated addObject:[NSNumber numberWithInteger: currentActivePlayers.count - 1]];
+    [self.delegate focusOnPlayerAtIndex:0];
+  }
+  
+  [self.delegate updateUIForPlayerAtIndex: playersIndexToUpdated];
+}
+
+-(void)handleDealerHandAfterPlayersDone {
+  [self.dealer.cards.lastObject setIsFaceUp:YES];
+  
+  while (self.dealer.cardsEvaluation <= DEALER_MINUMUM_CARD_EVALUATION) {
+    [self.dealer acceptNewCard:[self.deck drawRandomCardWithFaceUp:YES]];
+  }
+}
+
+-(void)handlePlayersBetsInRound {
+  for (Player *player in self.players) {
+    BOOL dealerBeatPlayer = (self.dealer.cardsEvaluation > player.cardsEvaluation && self.dealer.cardsEvaluation <= CARDS_AMOUNT_TO_WIN) || (self.dealer.state == BUST && player.state == BUST);
```

Comment:
These 2 booleans here looks really bad and really hard to read.

When seen such booleans, think if you can simplify the check - in your case, check if the dealer has a bust before

Reply:
Please break this whole logic better